src/eval.c

```diff
@@ -639,7 +710,8 @@ NULL
                 addReply(c,shared.czero);
         }
     } else if (c->argc == 3 && !strcasecmp(c->argv[1]->ptr,"load")) {
-        sds sha = luaCreateFunction(c,c->argv[2]);
+        /* In SCRIPT LOAD we pass in evalsha = 1 as a hint. */
+        sds sha = luaCreateFunction(c, c->argv[2], 1);
```

Comment:
shouldn't this be ?
```suggestion
        sds sha = luaCreateFunction(c, c->argv[2], 0);
```

Reply:
this is SCRIPT LOAD, we should pass in evalsha = 1.

Reply:
ohh, right :facepalm: